### ISSUE TITLE
Refactor CoursesController into multiple controllers

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#1417]
`CoursesController` is rather large, because it is responsible for a bunch of different independent things.

It could be reorganized into multiple smaller controllers. `#update` in particular has a lot of helper methods that are not reused by other actions, so most of the other actions could probably be moved to new controllers (or vice versa). One possible approach to doing that is described here: http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/

**Comment by testa19** [comment]
@ragesoss  Is it just another file `course_update_controller.rb` (`class CourseUpdateController < ApplicationController`)? 
(There would be nearly 170 and 100 LOC in two course-controller files if other update actions will be moved)

or is there will be some kind of Courses namespase (with folders)? CoursesController and Courses::UpdateController

Do i get the article right: other suggests controllers have to have only CRUD actions and any other actions could be moved to another controller and "translated" to CRUD there?

**Comment by ragesoss** [comment]
@testa19 I don't have set answers to those questions. If you're going to take the CRUD-only strategy from the blog post, I think that would involve leaving update (and create, and destroy, and show), adding a namespace, and creating new namespaced controllers for each of the other remaining actions.

On the other hand, separating update into its own controller makes the most sense in terms of single responsibility and isolating the most complex action into it's own file. I that case, Courses::UpdateController would probably be the best way to go. Doing it that way might help set up a pattern we can use for other controllers as well to start to simplify routes.rb, which is getting pretty messy without how far we've gone so far with most routes being explicitly defined.

If you want to take this on, you can play around with it and report what you discover. I'm not set on a particular solution, but the problem to solve here is that CoursesController is confusing because it does too many things.

**Comment by Aeropio** [comment]
hello @ragesoss, is this issue still open?

**Comment by ragesoss** [comment]
Yes.

**Comment by chepyego** [comment]
Is this issue still open? i would like to help out

**Comment by chepyego** [comment]
@ragesoss is this issue stil available?

**Comment by ragesoss** [comment]
yes, it is. if you're working on it, please just make a note of it here, and post an update if you stop working on it.

**Comment by chepyego** [comment]
Yes am working on it,, I'll update you

**Comment by chepyego** [comment]
hey, am sorry for not giving out feedback about the issue was to work on
due to COVID-19 schools were closed so getting internet has been a problem
to me and i see deadline is near ,I don't know how to start will you please
give some tips on how to go with this issue

**Comment by keshavg2** [comment]
Hi, @ragesoss these are the two helper methods update_courses_wikis and update_course_wiki_namespaces are used in both update and create methods in the courses controller. so I have a question about where we have to move these methods that'll be following the principle of DRY.

**Comment by ragesoss** [comment]
One option would be to extract them to a separate class, such a new service responsible just for updating those two things.

**Comment by Abdulwahab03** [comment]
hello, @ragesoss I would like to take this up

**Comment by ragesoss** [comment]
@Abdulwahab03 okay! please do this in as small and simple pieces as possible. the priority is to make the code easier to understand without breaking anything.

**Comment by Abdulwahab03** [comment]
thank you @ragesoss, I would like to ask for more guidance on this issue, perhaps I would like to check the previous PRs made just to serve as a guideline and make corrections on them, do you think this can help?
